### PR TITLE
Update docs

### LINKED ISSUE #111: Docs don't include all available loss functions
So the non-included funcitons are a little hard to find. CrossentropyLoss and PoissonLoss are examples.

### PR DESCRIPTION
**Pull request by juliohm**
This PR updates the docs to use Project.toml and a newer version of Documenter.jl. It also adds a new subsection covering the other losses available that do not fit the distance-based nor margin-based definition.

Fix #111

### CODE REVIEW

**Comment by juliohm** [comment]
It may depend on merging #124 first. Appreciate if you can review.

File: docs/Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+uuid = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
+
+[[Logging]]
+uuid = "56ddb016-857b-54e1-b83d-db4d58db5568"
+
+[[LossFunctions]]
+deps = ["InteractiveUtils", "LearnBase", "Markdown", "Random", "RecipesBase", "SparseArrays", "Statistics", "StatsBase", "Test"]
+git-tree-sha1 = "08d87fec43e7d335811dfae5b55dbfc5690e915b"
+uuid = "30fc2ffe-d236-52d8-8643-a9d8f7c094a7"
+version = "0.5.1"
```
Comment (oxinabox) [inline docs/Manifest.toml:70]: Need to do `] dev ..` so as to add currently version.
Otherwise dev version of docs will not be up to date

Comment (juliohm) [reply docs/Manifest.toml:70]: Maybe we should merge #124 first? I could then rebase this PR to use the new Project.toml, and then `dev ..` inside the docs folder.
